docs/index.md

```diff
@@ -1,94 +1,217 @@
+
+- [DirectXMath](...link...) As part of the Windows SDK will be the math library we use
+
+- [Assimp](www.assimp.org) Assimp provides facilities to load mesh files of various file formats
+
+- [dear IMGUI](https://github.com/ocornut/imgui) dear IMGUI is an easy to use immediate mode graphics user interface library, its widely used in the industry and we will be using it to display interactive options
+
+#### Project Layout
+
+- `.github` contains github related files to enable this project's existence
+- `docs` contains this very documentation describing each chapter
+- `lib` contains all external/third party libraries like `GLFW` or `AssImp`
+- `src` contains all the example source code behind this tutorial series
+
+# 0. Initial Setup
+
+```todo
+either clone repo, open empty solution with project and empty main.cpp
+or explain how to create a project in VS, set it up so that it makes sense,
+explain include/lib folders, how to set output folder and why Win32 is not
+an option really anymore either
+```
+
+Assume we cloned the empty main, already setup with GLFW
+
+Ok let's start with the whole example for a HelloWindow first
+
+```cpp
+#include <GLFW/glfw3.h>
+#include <cstdint>
+#include <iostream>
+
+int main(int argc, char* argv[])
+{
+    if (!glfwInit())
+    {
+        std::cout << "GLFW: Unable to initialize\n";
+        return -1;
+    }
+
+    const GLFWmonitor* primaryMonitor = glfwGetPrimaryMonitor();
+    const GLFWvidMode* videoMode = glfwGetVideoMode(primaryMonitor);
+    const int32_t width = static_cast<int32_t>(videoMode->width * 0.9f);
+    const int32_t height = static_cast<int32_t>(videoMode->height * 0.9f);
+
+    glfwWindowHint(GLFW_SCALE_TO_MONITOR, GLFW_FALSE);
+    glfwWindowHint(GLFW_CLIENT_API, GLFW_NO_API);
+    const GLFWwindow* window = glfwCreateWindow(
+        width,
+        height,
+        "LearnD3D11 - Hello Window",
+        nullptr,
+        nullptr);
+    if (window == nullptr)
+    {
+        std::cout << "GLFW: Unable to create window\n";
+        glfwTerminate();
+        return -1;
+    }
+
+    const int32_t windowLeft = videoMode->width / 2 - width / 2;
+    const int32_t windowTop = videoMode->height / 2 - height / 2;
+    glfwSetWindowPos(window, windowLeft, windowTop);
+
+    while (!glfwWindowShouldClose(window))
     {
-        public SomeCSharpCodeHere()
-        {
-        }
+        glfwPollEvents();
+        // future update code
+        // future render code
     }
-    ```
 
-    ```xml
-    <Why Not="Some">
-        <Xml/>
-    </Why>
-    ```
+    glfwDestroyWindow(window);
+    glfwTerminate();
+    return 0;
+}
+```
+
+Now let me explain what all these things mean.
+
+```cpp
+#include <GLFW/glfw3.h>
+```
+
+C++ needs to know where all the definitions and declarations are coming from.
```

Comment:
"C++ needs to know where all the definitions and declarations are coming from." is ambiguous and does not really specifically explain the reason of #include<GLFW/glfw3.h> specifically, I recommend modification of this phrase to:

"C++ needs to know where all the definitions and declarations of GLFW are coming from; `#include<GLFW/glfw3.h>` allows you to include the file that contains all of these definitions for GLFW."

Reply:
thats why there is a ```todo box right above the code under 0. Initial Setup

Reply:
ah, okay, sounds good